lib/amqp.js

```diff
@@ -125,26 +125,36 @@ class Amqp {
 
     async publishMessage(exchangeName, routingKey, payloadBuffer, options, iteration) {
         const settings = this.settings;
-        const result = this.publishChannel.publish(exchangeName, routingKey, payloadBuffer, options);
-        if (!result) {
-            log.warn('Buffer full when publishing a message to '
-                + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
+        const publishChannel = this.publishChannel;
+
+        function promisifiedPublish(exchangeName, routingKey, payloadBuffer, options) {
+            let result;
+            return new Promise((resolve, reject) => {
+                result = publishChannel.publish(exchangeName, routingKey, payloadBuffer, options, (err, ok) => {
+                    err ? reject(err) : resolve(ok);
+                });
+            })
+                .then(() => result);
         }
```

Comment:
You can make it shorter:
```
        const promisifiedPublish = (exchangeName, routingKey, payloadBuffer, options) =>
            new Promise((resolve, reject) => {
                const result = this.publishChannel.publish(exchangeName, routingKey, payloadBuffer, options, err => {
                    err ? reject(err) : resolve(result);
                });
            });
```

Reply:
I can not do this due to `stub` of this function in tests

in tests is use next `stub`:
```
(exchangeName, routingKey, payloadBuffer, options, cb) => 
{
  cb(null, 'Success');
  return true;
})
```
i call `calback` function and then do `return`

in your `function` when callback will be called `result` will be `undefined` so will work the next section of code https://github.com/elasticio/sailor-nodejs/blob/8c83b235e4528d430cd56505f19bce91a9f82c1b/lib/amqp.js#L119
what is not good.

Reply:
maybe you have advice about how to rewrite `stub` function

Reply:
I'm sure there is a way to write a test w/o trying to adopt code for a test case, but don't want to hold you with this task.